packages/go_router/test/go_router_test.dart

```diff
@@ -818,6 +819,19 @@ void main() {
               state.subloc == '/login' ? null : '/login');
 
       expect(router.location, '/login');
+
+      // Directly set the url through platform message.
+      final Map<String, dynamic> testRouteInformation = <String, dynamic>{
+        'location': '/dummy',
+      };
+      final ByteData message = const JSONMethodCodec().encodeMethodCall(
+        MethodCall('pushRouteInformation', testRouteInformation),
+      );
+      await ServicesBinding.instance.defaultBinaryMessenger
+          .handlePlatformMessage('flutter/navigation', message, (_) {});
```

Comment:
Consider making a helper for the platform message trick. I don't feel strongly about this, feel free to keep as-is if you prefer